hvac/tests/test_kubernetes_methods.py

```diff
@@ -0,0 +1,289 @@
+        expected_status_code = 200
+        mock_response = {
+            "auth": None,
+            "data": {
+                "keys": role_names,
+            },
+            "lease_duration": 0,
+            "lease_id": "",
+            "renewable": False,
+            "request_id": "e4c219fb-0a78-2be2-8d3c-b3715dccb920",
+            "warnings": None,
+            "wrap_info": None
+        }
+        mock_url = 'http://localhost:8200/v1/auth/{0}/role?list=true'.format(
+            'kubernetes' if mount_point is None else mount_point,
+        )
+        requests_mocker.register_uri(
+            method='GET',
+            url=mock_url,
+            status_code=expected_status_code,
+            json=mock_response,
+        )
+        client = Client()
+
+        test_arguments = dict()
+        if mount_point:
+            test_arguments['mount_point'] = mount_point
+        actual_response = client.list_kubernetes_roles(**test_arguments)
+
+        # ensure we received our mock response data back successfully
+        self.assertEqual(mock_response, actual_response)
+
+    @parameterized.expand([
+        ("default mount point", None, "application1"),
+        ("custom mount point", "k8s", "application2"),
+    ])
+    @requests_mock.Mocker()
+    def test_delete_kubernetes_role(self, test_label, mount_point, role_name, requests_mocker):
+        expected_status_code = 204
+        mock_url = 'http://localhost:8200/v1/auth/{0}/role/{1}'.format(
+            'kubernetes' if mount_point is None else mount_point,
+            role_name,
+        )
+        requests_mocker.register_uri(
+            method='DELETE',
+            url=mock_url,
+            status_code=expected_status_code,
+        )
+        client = Client()
+
+        test_arguments = dict(
+            role=role_name,
+        )
+        if mount_point:
+            test_arguments['mount_point'] = mount_point
+
+        actual_response = client.delete_kubernetes_role(**test_arguments)
+
+        self.assertEquals(
+            first=expected_status_code,
+            second=actual_response.status_code,
+        )
+
+    @parameterized.expand([
+        ("default mount point", "custom_role", "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9", None),
+        ("custom mount point", "custom_role", "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9", "gcp-not-default")
+    ])
+    @requests_mock.Mocker()
+    def test_auth_kubernetes(self, test_label, test_role, test_jwt, mount_point, requests_mocker):
+        mock_response = {
+            'auth': {
+                'accessor': 'accessor-1234-5678-9012-345678901234',
+                'client_token': 'cltoken-1234-5678-9012-345678901234',
+                'lease_duration': 10000,
+                'metadata': {
+                    'role': 'custom_role',
+                    'service_account_email': 'dev1@project-123456.iam.gserviceaccount.com',
+                    'service_account_id': '111111111111111111111'
```

Comment:
This mocked response does not look correct to me. The `service_account_email` and `service_account_id` fields are from the GCP auth backed and not the Kubernetes auth backend.

I believe the correct field names can be found in the Vault API docs at the below URL.

https://www.vaultproject.io/api/auth/kubernetes/index.html#sample-response-3

Reply:
I just pushed up the fix for this to your branch.